packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -128,7 +128,7 @@ export class ModuleGraph {
     for (const imported of importedModules) {
       const dep =
         typeof imported === 'string'
-          ? await this.ensureEntryFromUrl(imported, ssr)
```

Comment:
By this point, the imported URL is resolved, so using `ensureEntryFromUrl` is overkill. It pointlessly reruns the `resolveId` hook chain.